base/dates/parse.jl

```diff
@@ -0,0 +1,184 @@
+### Parsing utilities
+
+@generated function tryparse_internal{T<:TimeType, S, N}(::Type{T}, str::AbstractString, df::DateFormat{S, NTuple{N}}, raise::Bool=false)
+    token_types = Type[dp <: DatePart ? SLOT_RULE[first(dp.parameters)] : Void for dp in df.parameters[2].parameters]
+
+    types = slot_order(T)
+    num_types = length(types)
+    order = Vector{Int}(num_types)
+    for i = 1:num_types
+        order[i] = findfirst(token_types, types[i])
+    end
+
+    field_defaults = slot_defaults(T)
+    field_order = tuple(order...)
+    tuple_type = slot_types(T)
```

Comment:
@shashi it would only be an issue if `@ntuple` was forced to return an `NTuple`. Since it returns a `Tuple` we should be fine.